lib/MusicBrainz/Script/RemoveUnreferencedRows.pm

```diff
@@ -0,0 +1,112 @@
+package MusicBrainz::Script::RemoveUnreferencedRows;
+use Moose;
+
+use DBDefs;
+use MusicBrainz::Server::Context;
+use MusicBrainz::Server::Log qw( log_info );
+
+with 'MooseX::Runnable';
+with 'MooseX::Getopt';
+with 'MusicBrainz::Script::Role::Context';
+
+has dry_run => (
+    isa => 'Bool',
+    is => 'ro',
+    default => 0,
+    traits => [ 'Getopt' ],
+    cmd_flag => 'dry-run'
+);
+
+has summary => (
+    isa => 'Bool',
+    is => 'ro',
+    default => 1,
+);
+
+has verbose => (
+    isa => 'Bool',
+    is => 'ro',
+    default => 0,
+);
+
+sub run {
+    my ($self) = @_;
+
+    my ($count, $removed) = (0, 0);
+
+    log_info { 'Checking all unreferenced rows' };
+
+    my @unreferenced_rows = @{
+        $self->c->sql->select_list_of_lists(<<~'SQL');
+              SELECT table_name,
+                     array_agg(row_id) AS row_ids
+                FROM unreferenced_row_log
+               WHERE inserted < now() - '7 day'::interval
+            GROUP BY table_name
+            SQL
+    };
+
+    for my $table_group (@unreferenced_rows) {
+        my ($table_name, $row_ids) = @$table_group;
+
+        log_info { sprintf "Checking rows for $table_name" };
+
+        # We want to ensure the rows are still unreferenced
+        my $query = <<~"SQL";
+            SELECT id
+              FROM $table_name
+             WHERE ref_count = 0
+               AND id = any(?)
+            SQL
+
+        my $unused_ids = $self->c->sql->select_single_column_array(
+            $query,
+            $row_ids,
+        );
+
+        for my $id (@$unused_ids) {
+            ++$count;
```

Comment:
AFAICT the count is incremented unconditionally here, so it's equivalent to `my $count = scalar @$unused_ids;`.